exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -248,95 +257,134 @@ protected void buildSchema() throws SchemaChangeException {
     }
   }
 
```

Comment:
It would be nicer if *buildSchema()* was calling the "standard" *prefetchFirstBatchFromBothSides()* like the other binary operators (sans MJ). Instead of having specialized different code, and modifying the RecordBatch just for the hash join.